src/js/contexts/AnalyticsContext/AnalyticsContext.js

```diff
@@ -0,0 +1,40 @@
+import React, { useCallback, useContext, useEffect, useRef } from 'react';
+
+export const AnalyticsContext = React.createContext(() => {});
+
+export const useAnalytics = () => useContext(AnalyticsContext);
+
+export const AnalyticsProvider = ({ onAnalytics, children }) => {
+  const pathInfoRef = useRef({
+    lastUrl: window.location.href,
+    observer: undefined,
+  });
+
+  const sendAnalytics = useCallback(
+    (data) => onAnalytics && onAnalytics(data),
+    [onAnalytics],
+  );
+
+  useEffect(() => {
+    let observer;
+    if (onAnalytics) {
+      observer = new window.MutationObserver(() => {
+        const url = window.location.href;
+        if (url !== pathInfoRef.current.lastUrl) {
+          pathInfoRef.current.lastUrl = url;
+          sendAnalytics({ type: 'pageView', data: url });
```

Comment:
previousUrl would be useful in this event in order to get a sense of from and to.